examples/siro_sandbox/gui_navigation_helper.py

```diff
@@ -167,13 +174,28 @@ def _get_humanoid_walk_hints(
                 if visualize_path:
                     self._viz_path(path)
             else:
-                walk_dir = None
+                # TODO: potentially check for previous rotation
+                distance_multiplier = 1.0
+                # walk_dir = None
 
         if walk_dir is None and target_rot_quat is not None:
             walk_dir = self._compute_forward_dir(target_rot_quat)
             distance_multiplier = 0.0
 
-        return walk_dir, distance_multiplier
+        if (
+            found_path
+            and len(path.points) >= 2
+            and path.geodesic_distance >= dist_to_always_move_forward
+        ):
+            target_rot_quat = None
+
+        if target_rot_quat is not None:
+            # Get the forward direction
+            forward_gaze = target_rot_quat.transform_vector(
+                mn.Vector3(0, 0, 1)
+            )
+            forward_gaze.y = 0
+        return walk_dir, distance_multiplier, forward_gaze
```

Comment:
Why do we need `walk_dir` *and* `forward_gaze`? Aren't these basically the same thing? They are a vector in the ground plane indicating the direction the agent is facing, right?

Reply:
Not exactly. When we move backwards, the forward gaze and the walk_dir will be opposite of each other. The forward gaze is where we should be looking whereas forward dir is where we should be walking towards. I can add a comment